### PR TITLE
Add support for ephemeral P3A metrics, add ads enabled installation time metric

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves brave/brave-browser#28109

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

0. Start browser with fresh profile. Ensure metric does not exist in local state page.
1. Enable ads, ensure metric has value of 0.
2. Use MITM proxy to ensure metric is sent, ensure metric is marked as sent on local state page.
3. Close browser, move system time ahead by one calendar month, launch browser. Ensure metric is no longer present on local state page.
4. Create new profile, enable ads, ensure metric does not appear on local state page
5. Retest the above steps, but move the system time ahead by various amounts before step 1, in order to test different metric values.

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
LGTM

File: browser/brave_browser_process_impl.cc [hunk 1]
```diff
@@ -130,6 +127,9 @@ BraveBrowserProcessImpl::BraveBrowserProcessImpl(StartupData* startup_data)
   histogram_braveizer_ = brave::HistogramsBraveizer::Create();
 #endif  // BUILDFLAG(BRAVE_P3A_ENABLED)
 
+  // initialize ads stats helper
+  ads_brave_stats_helper();
```
Comment (iefremov) [inline browser/brave_browser_process_impl.cc:131]: are you sure you want to change the order of initialization?

File: components/p3a/metric_names.h [hunk 2]
```diff
@@ -177,14 +177,22 @@ constexpr inline auto kCollectedSlowHistograms =
...
-    "Brave.P3A.ExampleExpressMetric"
+    // Keep the top metric for the express unit test
+    // Remove it when there are other non ephemeral metrics at the beginning of the set
+    "Brave.P3A.TestExpressMetric",
+    "Brave.Rewards.EnabledInstallationTime"
+});
+
+// List of metrics that should only be sent once per latest histogram update.
+// Once the metric value has been sent, the value will be removed from the log store.
+constexpr inline auto kEphemeralHistograms =
```
Comment (iefremov) [inline components/p3a/metric_names.h:193]: we need to start spec'ing all these details, b/c new concepts like ephemeral metrics are not really clear

**Review by iefremov — APPROVED** [review]

File: browser/brave_browser_process_impl.cc (hunk 1 above)
Comment (DJAndries) [reply browser/brave_browser_process_impl.cc:131]: nope, that was a mistake i made during a rebase. thanks for pointing that out

File: components/p3a/metric_names.h (hunk 2 above)
Comment (DJAndries) [reply components/p3a/metric_names.h:193]: yes, 100% agree. i have a comprehensive & technical P3A document in the works

**Pull request merged**